src/tests/performance/connections.rs

```diff
@@ -173,16 +246,48 @@ async fn p002_connections_load() {
         all_stats.push(stats);
     }
 
-    // TODO: assertions with expected test results.
-
     node.stop().expect("unable to stop the node");
 
     // Display results table
     println!("\r\n{}", fmt_table(Table::new(&all_stats)));
+
+    // Check that results are okay
+    for stats in all_stats.iter() {
+        // No connection should be terminated.
+        assert_eq!(stats.terminated, 0, "Stats: {:?}", stats);
+
+        // We expect to have `MAX_PEERS` connections. This is only true if
+        // `stats.terminated == 0`.
+        assert_eq!(stats.accepted, MAX_PEERS, "Stats: {:?}", stats);
+
+        // The rest of the peers should be rejected.
+        assert_eq!(
+            stats.rejected,
+            stats.peers - MAX_PEERS,
+            "Stats: {:?}",
+            stats
+        );
+
+        // And no connection timeouts or errors
+        assert_eq!(stats.timed_out, 0, "Stats: {:?}", stats);
+        assert_eq!(stats.conn_error, 0, "Stats: {:?}", stats);
+    }
 }
 
-async fn simulate_peer(node_addr: SocketAddr, handshake_complete: Sender<()>) {
-    let config = TestConfig::default();
+async fn simulate_peer(node_addr: SocketAddr, handshake_complete: Sender<()>, thread_num: usize) {

```

Comment:
we could also rename `thread_num` to something else here

Reply:
Changed.